algoliasearch-common/src/main/java/com/algolia/search/Index.java

```diff
@@ -1433,44 +1433,74 @@ default Task batchRules(@Nonnull List<Rule> rules, @Nonnull RequestOptions reque
 
   /**
    * Delete records matching a query, with a batch size of 1000, internally uses browse
+   * Deprecated, use deleteBy
    *
    * @param query The query
    * @throws AlgoliaException
    */
+  @Deprecated
   default void deleteByQuery(@Nonnull Query query) throws AlgoliaException {
     deleteByQuery(query, RequestOptions.empty);
   }
 
   /**
+   * Delete records matching a query
+   *
+   * @param query The query
+   * @throws AlgoliaException
+   */
+  default Task deleteBy(@Nonnull Query query) throws AlgoliaException {
+    return deleteBy(query, RequestOptions.empty);
+  }
+
+  /**
    * Delete records matching a query, with a batch size of 1000, internally uses browse
+   * Deprecated, use deleteBy
    *
    * @param query          The query
    * @param requestOptions Options to pass to this request
    * @throws AlgoliaException
    */
+  @Deprecated
   default void deleteByQuery(@Nonnull Query query, @Nonnull RequestOptions requestOptions) throws AlgoliaException {
     getApiClient().deleteByQuery(getName(), query, 1000, requestOptions);
   }
 
   /**
+   * Delete records matching a query
+   * Deprecated, use deleteBy
+   *
+   * @param query          The query
+   * @param requestOptions Options to pass to this request
+   * @throws AlgoliaException
+   */
+  default Task deleteBy(@Nonnull Query query, @Nonnull RequestOptions requestOptions) throws AlgoliaException {
+    return getApiClient().deleteBy(getName(), query, requestOptions);
+  }
+
+  /**
    * Delete records matching a query, internally uses browse
+   * Deprecated use deleteBy
```

Comment:
Doesn't Javadoc have a `@deprecated` tag? I know we already have an annotation, but if we put something in the comment, why not use it?